reactor-core/src/main/java21/reactor/core/scheduler/LoomBoundedElasticScheduler.java

```diff
@@ -0,0 +1,1394 @@
+			this.idleSinceTimestamp = -1L;
+			MARK_COUNT.set(this, now ? DISPOSED_NOW_STATE : DISPOSED_STATE);
+			int previousState = markShutdown(this);
+
+			if (previousState != 0) {
+				return;
+			}
+
+			drain();
+		}
+
+		/**
+		 * An alias for {@link #release()}.
+		 */
+		@Override
+		public void dispose() {
+			this.release();
+		}
+
+		@Override
+		public Disposable schedule(Runnable task) {
+			return schedule(task, false);
+		}
+
+		Disposable schedule(Runnable task, boolean direct) {
+			Objects.requireNonNull(task, "Task should be non-null");
+			ensureQueueCapacityAndAddTasks(1);
+
+			Runnable decoratedTask = onSchedule(task);
+			ScheduledDisposable disposable =
+					new ScheduledDisposable(this, decoratedTask, 0, TimeUnit.NANOSECONDS, direct);
+
+			this.tasksQueue.offer(disposable);
+
+			trySchedule();
+
+			return disposable;
+		}
+
+		@Override
+		public Disposable schedule(Runnable task, long delay, TimeUnit unit) {
+			return schedule(task, delay, unit, false);
+		}
+
+		Disposable schedule(Runnable task, long delay, TimeUnit unit, boolean direct) {
+			Objects.requireNonNull(task, "Task should be non-null");
+			Objects.requireNonNull(unit, "TimeUnit should be non-null");
+			if (delay <= 0) {
+				throw new IllegalArgumentException();
+			}
+
+			ensureQueueCapacityAndAddTasks(1);
+
+			Runnable decoratedTask = onSchedule(task);
+			ScheduledDisposable disposable =
+					new ScheduledDisposable(this, decoratedTask, 0, unit, direct);
+
+			disposable.schedule(delay, unit);
+
+			return disposable;
+		}
+
+		@Override
+		public Disposable schedulePeriodically(Runnable task,
+				long initialDelay,
+				long period,
+				TimeUnit unit) {
+			return schedulePeriodically(task, initialDelay, period, unit, false);
+		}
+
+		Disposable schedulePeriodically(Runnable task,
+				long initialDelay,
+				long period,
+				TimeUnit unit,
+				boolean direct) {
+			Objects.requireNonNull(task, "Task should be non-null");
+			Objects.requireNonNull(unit, "TimeUnit should be non-null");
+			if (initialDelay <= 0 || period <= 0) {
+				throw new IllegalArgumentException();
+			}
+
+			ensureQueueCapacityAndAddTasks(1);
+
+			Runnable decoratedTask = onSchedule(task);
+			ScheduledDisposable disposable =
+					new ScheduledDisposable(this, decoratedTask, period, unit, direct);
+
+			disposable.schedule(initialDelay, unit);
+
+			return disposable;
+		}
+
+		void trySchedule() {
+			int previousState = addWork(this);
+			if (previousState != 0) {
+				return;
+			}
+
+			drain();
+		}
+
+		void tryForceSchedule() {
+			int previousState = forceAddWork(this);
+			if (previousState != 0) {
+				return;
+			}
+
+			drain();
+		}
+
+		void drain() {
+			final Queue<ScheduledDisposable> q = this.tasksQueue;
+
+			int m = this.state;
+			for (;;) {
+				for (;;) {
+					if (isDisposedNow(this.markCount)) {
+						clearAllTask();
+						countDown();
+						return;
+					}
+
+					final ScheduledDisposable task = q.poll();
+
+					if (task == null) {
```

Comment:
This can potentially lead to freeing up the queue and removing the boundedState if no Worker has a reference any more.